File: 2025/2025_04/Pro_389479.java
class Solution {
    static int[] servers;
    public int solution(int[] players, int m, int k) {
       servers = new int[players.length];
        int answer = 0;
        for(int i=0;i<players.length;i++){
            int serverCnt=0;
            
            if(players[i]/m>0){
                serverCnt = players[i]/m - servers[i];
            }
            

            if(serverCnt>0){
                answer+=serverCnt;
                for(int j=i;j<i+k&&j<players.length;j++){
                    servers[j]+=serverCnt;
                }
                
            }
            
        }
        return answer;
    }
}

File: 2025/2025_04/Pro_389479.cpp
#include <bits/stdc++.h>

using namespace std;
int serverCount(int m, int num)
{
    for (int i = 0; i <= 1000; i++)
    {
        if ((i * m) <= num && num < (i + 1) * m)
        {
            return i;
        }
    }
    return 0;
}

int solution(vector<int> players, int m, int k)
{
    int answer = 0; // 증설횟수
    vector<int> servers(players.size());
    for (int i = 0; i < players.size(); i++)
    {
        int num = players[i];
        int serverCnt = serverCount(m, num);
        if (servers[i] <= serverCnt)
        {

            int newServerCnt = serverCnt - servers[i];
            answer += newServerCnt;
            for (int j = i; j < i + k && j < players.size(); j++)
            {
                servers[j] += newServerCnt;
            }
        }
    }

    return answer;
}